src/meta.rst

```diff
@@ -71,39 +71,128 @@ A full description of selectors is
+
+Note that ``boost`` and ``numpy`` are exceptions to this. See ``Building Against NumPy`` section.
+
+
+Build matrices
+---------------------
+
+``conda-forge.yml``'s build matrices is removed in conda-smithy=3. To get a build matrix, create a ``conda_build_config.yaml`` file inside recipe folder. For example following will give you 2 builds and you can use the selector ``vtk_with_osmesa`` in the ``meta.yaml``
+
+.. code-block:: yaml
+
+    vtk_with_osmesa:
+      - False
+      - True
+
+You need to rerender the feedstock after this change.
+
+Currently, ``python, vc, r-base`` will create a matrix of jobs for each supported version. If ``python`` is only a build dependency and not a runtime dependency (eg: build script of the package is written in Python, but the package is not dependent on python), use ``build`` section
+
+Following implies that ``python`` is only a build dependency and no Python matrix will be created.
+
+.. code-block:: yaml
+
+    build:
+      - python
+    host:
+
+
+Following implies that ``python`` is a runtime dependency and a Python matrix for each supported python version will be created.
+
+.. code-block:: yaml
+
+    host:
+      - python
+
+
+
+Compilers
+---------
+
+Although ``conda-build=3`` gives the ability to use Anaconda 5 compilers, they should be used only on Windows as packages built with them are incompatible with conda-forge packages built using the CI compilers. All the packages in conda-forge needs to be rebuilt using the compilers and released simultaneously to avoid package incompatibility.
+
+Following is suggested presently,
+
+.. code-block:: yaml
+
+    requirements:
+      build:
+        - {{ compiler('cxx') }}     # [win]
+        - toolchain                 # [unix]
+
+but in the future you should switch to ``{{`compiler('cxx') }}``. An announcement will be made when this should be done.
+
+Reason for holding out using the Anaconda 5 compilers is that the packages built by them are sometimes incompatible with the packages built with the older compilers in CI platforms conda-forge has been using. All the dependencies of a package should be compiled (This is not true for all cases, but it's better to be cautious) with the new compilers before using the new compiler in a package. This presents a problem in that rebuilding a package will break the dependent packages. Therefore, ``conda-forge`` has decided to rebuild all of the packages and upload them all at once. More details on how this is done will be communicated in the future.
 
 
 Building Against NumPy
 ----------------------
 If you have a package which links\* against ``numpy`` you can build against the oldest possible version of ``numpy`` that is forwards compatible.
-That can be achieved by pinning the build requirements and letting "free" the run requirements.
+That can be achieved by pinning the host requirements and letting "free" the run requirements.
```

Comment:
Replace this sentence with:

With conda-build 3, we can leave the pin empty for build-time, and conda-build will use the ``numpy`` key from conda_build_config.yaml.  We can also utilize conda-build's dynamic pinning with its ``pin_compatible`` function to evaluate the numpy pin based on the version that actually gets used at build time.